Assets/Scripts/UI/GalleryMenu.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using Newtonsoft.Json;
+using Unity.VisualScripting;
+using UnityEditor;
+namespace UI
+{
+    public class GalleryMenu : MonoBehaviour
+    {
+
+        private List<RectTransform> gridElements;
+        private List<GameObject> unlockedElements;
+        private int pageIndex = 0;
+
+        public Transform grid;
+        public GameObject gridElement;
+        private void Start()
+        {
+            for(int i = 0; i < 15; i++)
+                Instantiate(gridElement, grid);
```

Comment:
You could've built the list of grid elements from the return values of the Instantiate calls...